### PR TITLE
num_sample variation fixes

### PR DESCRIPTION
**Pull request by benidis**
Fixes issue #301 (following #232).

### CODE REVIEW

**Comment by lostella** [comment]
Looks like this broke the tests for `shell`: looking at the logs it seems that the `num_samples` configuration is not catched and 100 samples (the default somewhere?) get produced instead of 4

File: src/gluonts/model/forecast.py [hunk 1]
```diff
@@ -613,7 +613,7 @@ class OutputType(str, Enum):
 
 
 class Config(pydantic.BaseModel):
-    num_eval_samples: int = pydantic.Schema(100, alias="num_samples")
+    num_samples: int = pydantic.Schema(100, alias="num_samples")
```
Comment (jaheba) [inline src/gluonts/model/forecast.py]: To ensure backwards compatibility, this should be:

```suggestion
    num_samples: int = pydantic.Schema(100, alias="num_eval_samples")
```

File: src/gluonts/model/npts/_predictor.py [hunk 2]
```diff
@@ -169,11 +168,11 @@ def __init__(
         use_default_time_features: bool = True,
         num_default_time_features: int = 1,
         feature_scale: float = 1000.0,
-        num_parallel_samples: int = 100,
+        num_samples: int = 100,
```
Comment (lostella) [inline src/gluonts/model/npts/_predictor.py]: My proposal was to remove this from the constructor of predictors, and keep it only as an argument to the `predict` method, see #301

File: src/gluonts/model/prophet/_predictor.py [hunk 3]
```diff
@@ -140,34 +137,37 @@ def __init__(
 
         assert "uncertainty_samples" not in prophet_params, (
             "Parameter 'uncertainty_samples' should not be set directly. "
-            "Please use 'num_eval_samples' instead."
+            "Please use 'num_samples' instead."
```
Comment (lostella) [inline src/gluonts/model/prophet/_predictor.py]:
```suggestion
            "Please use 'num_samples' in the `predict` method instead."
```

**Pull request merged**